portal/models/communication.py

```diff
@@ -252,6 +252,34 @@ def generate_and_send(self):
         self.message.send_message()
         self.status = 'completed'
 
+    def preview(self):
```

Comment:
Can we avoid repeating such similar code blocks between `generate_and_send` and `preview`?  (i.e. both use a common block?)

Reply:
Sure thing - I've moved that repeated code into a separate function.

Reply:
@pbugni (whenever you get a chance, this is ready for another review)